src/main/java/com/webtutsplus/ecommerce/service/CartService.java

```diff
@@ -0,0 +1,76 @@
+package com.webtutsplus.ecommerce.service;
+
+import com.webtutsplus.ecommerce.dto.CartDto;
+import com.webtutsplus.ecommerce.dto.ProductDto;
+import com.webtutsplus.ecommerce.model.*;
+import com.webtutsplus.ecommerce.repository.CartRepository;
+import com.webtutsplus.ecommerce.repository.WishListRepository;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import javax.xml.crypto.Data;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+@Service
+@Transactional
+public class CartService {
+
+    private final CartRepository cartRepository;
+
+    public CartService(CartRepository cartRepository) {
+        this.cartRepository = cartRepository;
+    }
+
+    public static Cart getCartFromDto(CartDto cartDto, Product product,int userId) {
+        Cart cart = new Cart(cartDto, product,userId);
+        return cart;
+    }
+
+    public void addToCart(CartDto cartDto, Product product,int userId) {
+        Cart cart = getCartFromDto(cartDto, product,userId);
+        cartRepository.save(cart);
+    }
+
+    public CartCost listCartItems(int user_id) {
+        List<Cart> cartList = cartRepository.findAllByUserIdOrderByCreatedDateDesc(user_id);
+        List<CartDto> cartItems = new ArrayList<>();
+        for (Cart cart:cartList){
+            CartDto cartDto = getDtoFromCart(cart);
+            cartItems.add(cartDto);
+        }
+        int totalCost = 0;
```

Comment:
should be double, as price can be double

Reply:
solved